src/main/java/io/cryostat/agent/WebServer.java

```diff
@@ -93,62 +107,16 @@ void start() throws IOException, NoSuchAlgorithmException {
-                                String mtd = exchange.getRequestMethod();
-                                switch (mtd) {
-                                    case "POST":
-                                        synchronized (WebServer.this.credentials) {
-                                            executor.execute(registration.get()::tryRegister);
-                                            exchange.sendResponseHeaders(
-                                                    HttpStatus.SC_NO_CONTENT, -1);
-                                            exchange.close();
-                                        }
-                                        break;
-                                    case "GET":
-                                        exchange.sendResponseHeaders(HttpStatus.SC_NO_CONTENT, -1);
-                                        exchange.close();
-                                        break;
-                                    default:
-                                        exchange.sendResponseHeaders(HttpStatus.SC_NOT_FOUND, -1);
-                                        exchange.close();
-                                        break;
-                                }
-                            }
-                        });
-        pingCtx.setAuthenticator(new AgentAuthenticator());
-        pingCtx.getFilters()
-                .add(
-                        new Filter() {
-                            @Override
-                            public void doFilter(HttpExchange exchange, Chain chain)
-                                    throws IOException {
-                                long start = System.nanoTime();
-                                String requestMethod = exchange.getRequestMethod();
-                                String path = exchange.getRequestURI().getPath();
-                                log.info("{} {}", requestMethod, path);
-                                chain.doFilter(exchange);
-                                long elapsed = System.nanoTime() - start;
-                                log.info(
-                                        "{} {} : {} {}ms",
-                                        requestMethod,
-                                        path,
-                                        exchange.getResponseCode(),
-                                        Duration.ofNanos(elapsed).toMillis());
-                            }
-
-                            @Override
-                            public String description() {
-                                return "requestLog";
-                            }
-                        });
+        Set<RemoteContext> mergedContexts = new HashSet<>(remoteContexts.get());
+        mergedContexts.add(new PingContext());
+        mergedContexts.forEach(
+                rc -> {
+                    HttpContext ctx = this.http.createContext(rc.path(), rc::handle);
+                    ctx.setAuthenticator(new AgentAuthenticator());
```

Comment:
Should this just use the same AgentAuthenticator instance instead of creating a `new` one for each context?